### PR TITLE
feat(tool): add analysis for delayed receipts

### PR DESCRIPTION
**Pull request by bowenwang1996**
Add a script that analyzes congestion of shards by looking at delayed receipts for each shard.

### CODE REVIEW

**Review by saketh-are — APPROVED** [review]

File: tools/database/src/analyze_delayed_receipt.rs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        let node_storage = open_storage(&home, &mut near_config).unwrap();
+        let store = node_storage.get_split_store().unwrap_or_else(|| node_storage.get_hot_store());
+        let chain_store = Rc::new(ChainStore::new(
+            store.clone(),
+            near_config.genesis.config.genesis_height,
+            false,
+        ));
+        let shard_layout = ShardLayout::get_simple_nightshade_layout();
+        // hard code the shard ids
+        let shard_ids = vec![0, 1, 2, 3];
```
Comment (wacban) [inline tools/database/src/analyze_delayed_receipt.rs:48]: nit: would be nicer to get the shard layout and shard ids for each block individually then you always get the right one and you also don't need to handle the shard layout changing

File: tools/database/src/analyze_delayed_receipt.rs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+use near_primitives::shard_layout::{ShardLayout, ShardUId};
+use near_primitives::types::BlockHeight;
+
+use crate::block_iterators::{
+    make_block_iterator_from_command_args, CommandArgs, LastNBlocksIterator,
+};
+use nearcore::open_storage;
+
+#[derive(Parser)]
+pub(crate) struct AnalyzeDelayedReceiptCommand {
```
Comment (wacban) [inline tools/database/src/analyze_delayed_receipt.rs:22]: Can you add a small comment?

File: tools/database/src/analyze_delayed_receipt.rs [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        if let Some((block_height, block_hash)) = first_analysed_block {
+            println!("Block: height = {block_height}, hash = {block_hash}");
+        }
+        if let Some((block_height, block_hash)) = last_analysed_block {
+            println!("Block: height = {block_height}, hash = {block_hash}");
+        }
+
+        for shard_id in shard_ids {
+            let congested_chunks = *shard_id_to_congested.get(&shard_id).unwrap_or(&0);
+            println!("shard {} congested ratio {}", shard_id, congested_chunks as f64 / blocks_count as f64);
```
Comment (wacban) [inline tools/database/src/analyze_delayed_receipt.rs:118]: mini nit: divide by zero

**Review by wacban — APPROVED** [review]
LGTM

It's perfect as is for a one off tool so please consider all comments optional.

File: tools/database/src/analyze_delayed_receipt.rs (hunk 1 above)
Comment (wacban) [reply tools/database/src/analyze_delayed_receipt.rs:48]: I think with epoch manager you can do something like the following:
block -> epoch id -> shard layout

**Pull request merged**